packages/scaffold-config/src/detect.ts

```diff
@@ -120,7 +120,22 @@ export function detectLanguage (projectRoot: string, pkgJson: PkgJson): 'js' | '
     debug('Did not find cypress.config file')
   }
 
-  if ('typescript' in (pkgJson.dependencies || {}) || 'typescript' in (pkgJson.devDependencies || {})) {
+  // If we can't find an installed TypeScript or ESBuild, there's no way we can assume the project is using TypeScript,
+  // because it won't work on the next step of installation anyway
+  try {
+    debug('Resolved typescript from %s', require.resolve('typescript', { paths: [projectRoot] }))
```

Comment:
nit, it seems a little weird to be doing the required operation within a debug statement. Not knowing what 'debug' does I don't know if require.resolve is executed or ignored when debug mode is disabled.

```suggestion
    const typescriptFile=  require.resolve('typescript', { paths: [projectRoot] })
    debug('Resolved typescript from %s', typescriptFile )
```

Reply:
Fair enough, yeah since it's an argument to `debug` there's nothing about being in debug mode that would change the evaluation of it, but it does read out a bit confusing.